src/main/java/org/hydev/mcpm/client/Downloader.java

```diff
@@ -67,4 +120,53 @@ public Downloader threads(int threads)
         this.threads = threads;
         return this;
     }
-}
+
+    class Processor implements Runnable {
+
+        private List<String> files;
+        private Map<String, File> urls;
+        private int id;
+
+        /**
+         * Download multiple files from the internet to local storage through HTTP requests
+         * <p>
+         * The implementation must use multithreading
+         *
+         * @param files List of remote urls
+         * @param urls Mapping of remote urls to local file paths
+         * @param id id for each file
+         */
+        public Processor(int id, Map<String, File> urls, List<String> files) {
+            this.id = id;
+            this.urls = urls;
+            this.files = files;
+        }
+
+        // Process downloading each file from multithreading
+        @Override
+        public void run() {
+            String url = files.get(id);
+            File to = urls.get(url);
+            downloadFile(url, to);
+        }
+    }
+
+
+    /**
+     * Displays a demo for.
+     *
+     * @param args Arguments are ignored.
+     */
+    public static void main(String[] args) throws IOException {
+        // Remember to chang link to test
+        String link = "https://sd.blackball.lv/library/Introduction_to_Algorithms_Third_Edition_(2009).pdf";
+        File out = new File("./Introduction_to_Algorithms_Third_Edition.pdf");
+        String link1 = "https://www.iusb.edu/students/academic-success-programs/academic-centers-for-excellence/docs/Basic%20Math%20Review%20Card.pdf";
+        File out1 = new File("./Math.pdf");
+        Downloader downloader = new Downloader();
+        Map<String, File> urls = new HashMap<>();
+        urls.put(link, out);
+        urls.put(link1, out1);
+        downloader.downloadFiles(urls, true, 2);
```

Comment:
I tested this with three files and two threads, just to ensure that it doesn't slow down significantly when there are more than 2 files being downloaded, and it seems to work perfectly :)